conda/recipes/cuml/meta.yaml

```diff
@@ -62,7 +62,7 @@ requirements:
     - raft-dask ={{ minor_version }}
     - scikit-build >=0.13.1
     - setuptools
-    - treelite =3.0.1
+    - treelite =3.1.0
```

Comment:
```suggestion
    - treelite {{ treelite_version }}
```

This file has two lines which specify the `treelite` version.

To DRY this up, can we move this version specifier to `conda/recipes/cuml/conda_build_config.yaml` and update both lines in this file to look like this suggestion?

This would make it consistent with the `libcuml` recipe also.